test/e2e/basics.go

```diff
@@ -491,12 +491,12 @@ func testIgnorePaths(client kubernetes.Interface) kubetest.TestSuite {
 			Then: kubetest.Actions(
 				kubetest.ClientSucceeds(
 					client,
-					fmt.Sprintf(commandWithoutAuth, "/", 401, 401),
+					fmt.Sprintf(commandWithoutAuth, "/", 403, 403),
```

Comment:
This probably changed because the delegating kube authenticator allows anonymous.  I think this is change is reasonable and in keeping with other delegated authentication usages, but it's worth calling out to a maintainer here and seeing how to write up a release note about this change and how versions should be numbered.  

The kube delegated authentication (used by kube-controller-manager for instance) doesn't even offer a no-anonymous option.

Reply:
cc @ibihim for ^

Reply:
It will be interesting to know if anyone is using this without authz. I thought this is rbac proxy. So authz should be its main functionality. But maybe people use it differently?

Reply:
if authz is optional in kube-rbac-proxy params, then you need a way to disable anonymous auth and the spot is here: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/server/options/authentication.go#L286

Reply:
If I [read this ](https://github.com/brancz/kube-rbac-proxy/blob/fc1ca4f969941340a8adb66932bd64dc5773d37a/pkg/proxy/proxy.go#L83)correctly, there is no way to run kube-rbac-proxy without authz. @ibihim can you confirm?